src/Async.ts

```diff
@@ -1,12 +1,25 @@
 import { Effect } from './Effect'
+import { fx } from './Fx'
+import { schedule } from './Time'
+import { dispose } from './internal/disposable'
 
 type Run<A> = (abort: AbortSignal) => Promise<A>
 
 export class Async extends Effect('fx/Async')<Run<any>> { }
 
 export const run = <const A>(run: Run<A>) => new Async(run).returning<A>()
 
-export const sleep = (ms: number) => run(abort => new Promise<void>(resolve => {
-  const timeout = setTimeout(resolve, ms)
-  abort.addEventListener('abort', () => clearTimeout(timeout), { once: true })
-}))
+// TODO: Should this move to Time?
+// TODO: Should Sleep be a distinct effect? It might be possible
+// to implement it more simply there
```

Comment:
The more I think about this, the more I'm ok with it as is.  Maybe it'll become more clear later.